spec/forms/nqt_in_academic_year_after_itt_form_spec.rb

```diff
@@ -0,0 +1,281 @@
+        params: params
+      )
+    end
+
+    before { form.save }
+
+    context "when invalid" do
+      let(:journey) { additional_payments_journey }
+
+      let(:claim) { create(:claim, policy: Policies::EarlyCareerPayments) }
+
+      let(:params) do
+        ActionController::Parameters.new(
+          claim: {
+            nqt_in_academic_year_after_itt: nil
+          }
+        )
+      end
+
+      it "returns false" do
+        expect { expect(form.save).to be false }.not_to(
+          change { claim.eligibility.reload.nqt_in_academic_year_after_itt }
+        )
+      end
+    end
+
+    context "when valid" do
+      let(:params) do
+        ActionController::Parameters.new(
+          claim: {
+            nqt_in_academic_year_after_itt: true
+          }
+        )
+      end
+
+      context "when the current journey is not `additional-payments`" do
+        let(:journey) { Journeys::TeacherStudentLoanReimbursement }
+
+        let(:claim) { create(:claim, policy: Policies::EarlyCareerPayments) }
+
+        it "sets the nqt_in_academic_year_after_itt attribute" do
+          expect(claim.eligibility.nqt_in_academic_year_after_itt).to be true
+        end
+
+        it "does not set the induction as complete" do
+          expect(claim.eligibility.induction_completed).to be nil
+        end
+      end
+
+      context "when the current journey is `additional-payments`" do
+        let(:journey) { additional_payments_journey }
+
+        context "when the teacher has not passed the tid details check" do
+          let(:claim) do
+            create(
+              :claim,
+              :logged_in_with_tid,
+              details_check: false,
+              policy: Policies::EarlyCareerPayments
+            )
+          end
+
+          it "sets the nqt_in_academic_year_after_itt attribute" do
+            expect(claim.eligibility.nqt_in_academic_year_after_itt).to be true
+          end
+
+          it "does not set the induction as complete" do
+            expect(claim.eligibility.induction_completed).to be nil
+          end
+        end
+
+        context "when the teacher has passed the tid details check" do
+          context "when there is not an eligible induction" do
+            let(:claim) do
+              create(
+                :claim,
+                :logged_in_with_tid,
+                policy: Policies::EarlyCareerPayments,
+                details_check: true,
+                dqt_teacher_status: nil
+              )
+            end
+
+            it "sets the nqt_in_academic_year_after_itt attribute" do
+              expect(
+                claim.eligibility.nqt_in_academic_year_after_itt
+              ).to be true
+            end
+
+            it "does not set the induction as complete" do
+              expect(claim.eligibility.induction_completed).to be nil
+            end
+          end
+
+          context "when there is an eligible induction" do
+            let(:claim) do
```

Comment:
Can we use a factory for this please? As the data schema changes this will require continual maintenence.